cidr-filtering-cni

```diff
@@ -0,0 +1,252 @@
+
+    # filter chains
+    ip netns exec "$CNI_CONTAINERID" $(create_chain ${NFT_TABLE_NETDEV} ${NFT_TABLE} m-ingress) || exitWithError "Failed to add chain m-ingress"
+    ip netns exec "$CNI_CONTAINERID" $(create_chain ${NFT_TABLE_BRIDGE} ${NFT_TABLE} m-egress) || exitWithError "Failed to add chain m-egress"
+
+    # setup prerouting chain
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule ${NFT_TABLE_NETDEV} ${NFT_TABLE} ${NFT_INGRESS_CHAIN} ether type arp counter accept) || exitWithError "Failed to allow arp for ingress"
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule ${NFT_TABLE_NETDEV} ${NFT_TABLE} ${NFT_INGRESS_CHAIN} iifname $CNI_IFNAME counter jump m-ingress) || exitWithError "Failed to add rule: jump m-ingress"
+
+    # setup postrouting chain
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule ${NFT_TABLE_BRIDGE} ${NFT_TABLE} ${NFT_POSTROUTING_CHAIN} ether type arp counter accept) || exitWithError "Failed to allow arp for egress"
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule ${NFT_TABLE_BRIDGE} ${NFT_TABLE} ${NFT_POSTROUTING_CHAIN} oifname $CNI_IFNAME counter jump m-egress) || exitWithError "Failed to add rule: jump m-egress"
+
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${NFT_TABLE_NETDEV}" "${NFT_TABLE}" m-ingress counter meta mark set meta mark "&" 0xfffcffff) || exitWithError "Failed to add rule: set meta mark & 0xfffcffff"
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${NFT_TABLE_BRIDGE}" "${NFT_TABLE}" m-egress counter meta mark set meta mark "&" 0xfffcffff) || exitWithError "Failed to add rule: set meta mark & 0xfffcffff"
+
+    create_chains_for_filtering $policy_id ${NFT_TABLE_NETDEV} "ingress"
+    create_chains_for_filtering $policy_id ${NFT_TABLE_BRIDGE} "egress"
+
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${NFT_TABLE_NETDEV}" "${NFT_TABLE}" m-ingress meta mark "&" 0x00030000 == 0x00030000 counter return) || exitWithError "Failed to add rule: mark & 0x00030000 == 0x00030000 counter return"
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${NFT_TABLE_NETDEV}" "${NFT_TABLE}" m-ingress counter drop) || exitWithError "Failed to add rule: drop"
+
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${NFT_TABLE_BRIDGE}" "${NFT_TABLE}" m-egress meta mark "&" 0x00030000 == 0x00030000 counter return) || exitWithError "Failed to add rule: mark & 0x00030000 == 0x00030000 counter return"
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${NFT_TABLE_BRIDGE}" "${NFT_TABLE}" m-egress counter drop) || exitWithError "Failed to add rule: drop"
+
+    exitWithSuccess
+;;
+
+DEL)
+    echo "Delete $CNI_CONTAINERID" >> $logFile
+    rm -f /var/run/netns/"$CNI_CONTAINERID"
+;;
+
+VERSION)
+    echo "{"cniVersion":"0.3.1","supportedVersions":["0.1.0","0.2.0","0.3.0","0.3.1"]}" >&3
+;;
+
+*)
+    exit 0
```

Comment:
maybe add CHECK stub, and exit 1 for the rest ?
unless there are more commands other than those 4

Reply:
Good point. I'll exit with unrecognized CNI command.
We don't need to deal with CHECK, since that is introduced in version 0.4.0 (and we don't need to support that IIUC)